arkouda-docker/build_docker_image.py

```diff
@@ -0,0 +1,196 @@
+import argparse, enum, subprocess, sys
+from typing import Optional
+from enum import Enum
+
+class ImageType(Enum):
+    '''
+    The ImageType enum provides controlled vocabulary for the docker image type
+    '''
+    ARKOUDA_FULL_STACK = 'ARKOUDA_FULL_STACK'
+    ARKOUDA_SMP_SERVER = 'ARKOUDA_SMP_SERVER'
+    ARKOUDA_UDP_SERVER = 'ARKOUDA_UDP_SERVER'
+    CHAPEL_UDP = 'CHAPEL_UDP'
+
+def getImageFile(imageType: ImageType) -> str:
+    '''
+    Returns the Dockerfile per the image type
+
+    :param ImageType imageType: Docker image type enum    
+    :return: Dockerfile corresponding to the image type
+    :rtype: str
+    '''
+    if ImageType.ARKOUDA_FULL_STACK == imageType:
+        return 'arkouda-full-stack'
+    elif ImageType.ARKOUDA_SMP_SERVER == imageType:
+        return 'arkouda-smp-server'
+    elif ImageType.ARKOUDA_UDP_SERVER == imageType:
+        return 'arkouda-udp-server'
+    elif ImageType.CHAPEL_UDP == imageType:
+        return 'chapel-gasnet-udp'
```

Comment:
would it make more sense to just return `imageType.name`? it feels less prone to typos

Reply:
those are the docker file names, so there has to be some type of lookup. I use the enum 'cause I figure the user will be more likely to mis-type. If I mis-type, shame on me, LOL

Reply:
but, yeah, I can set the imageType.value to the docker file name, will do that

Reply:
@pierce314159 updated, take a look

Reply:
awesome! i love it